### PR TITLE
GEN-1469 | Detect pre-selected product from URL in widget flow

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Parse URL search params and detect if we have a pre-selected product, then skip product selection screen

## Justify why they are needed

- I refactored the server-side code a bit to make it easier to read the code

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3583** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3583?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3584** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3584?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3584?utm_source=stack-comment).

File: apps/store/src/features/widget/parseSearchParams.ts [hunk 1]
```diff
@@ -32,3 +35,32 @@ export const parseCustomerDataSearchParams = (
...
+  if (subType) updatedSearchParams.delete(SearchParam.ApartmentSubType)
+
+  const productName = getProductName(productType, subType)
+
+  return [productName || productType, updatedSearchParams]
+}
+
+// Legacy, used by Partner Widget
+// TODO: migrate to use our new product names directly in the URL
+const getProductName = (productType: string | null, subType: string | null): string | undefined => {
```
Comment (alebedev) [inline apps/store/src/features/widget/parseSearchParams.ts:57]: Should we expose this data in product data we get storefront?

Comment (robinandeer) [reply apps/store/src/features/widget/parseSearchParams.ts:57]: Since it's a temporary thing I think that won't be necessary, right?

**Review by alebedev — APPROVED** [review]

**Review by guilhermespopolin — APPROVED** [review]

**Comment by robinandeer** [comment]
### Merge activity

* **Nov 20, 9:29 AM**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3584) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**